plugins/module_utils/utils.py

```diff
@@ -1298,3 +1299,36 @@ def count_terms(self, terms, module_parameters):
             terms = [terms]
 
         return len(set(terms).intersection(module_parameters))
+
+
+class NautobotApiBase:
```

Comment:
Could we consolidate the API base into the NautobotGraphQL? It seems like it's simply to instantiate `pynautobot`.

Also, `api` should probably be a positional argument instead of defaulting to `None` if we keep it this way.

Reply:
I thought about this some as well. I'm not sure if we will find the need to use that API object in other places in the future as well, outside of just the Graphql part.